src/14_legendre.jl

```diff
@@ -0,0 +1,419 @@
+@doc raw"""
+    chebyshevT(n, x)
+
+Evaluate the Chebyshev polynomials of the first kind ``T_n(x)`` of order ``n`` at position
+``x``, defined by
+
+```math
+T_n(x)
+=
+\begin{cases}
+    1
+    & \text{for} \quad n = 0\,, \; x \in [-1,1]
+    \\
+    x
+    & \text{for} \quad n = 1\,, \; x \in [-1,1]
+    \\
+    2 x T_{n-1}(x) - T_{n-2}(x)
+    & \text{for} \quad n \geq 2\,, \; x \in [-1,1]
+    \,.
+\end{cases}
+```
+
+External links: [DLMF](https://dlmf.nist.gov/18.3.T1),
+[Wikipedia](https://en.wikipedia.org/wiki/Chebyshev_polynomials).
+"""
+function chebyshevT(n::Integer, x::Real)
+    if n < 0
+        throw(DomainError(n, "must be non-negative"))
+    end
+    if x < -1 || x > 1
+        throw(DomainError(x, "must be in the range [-1,1]"))
+    end
+
+    ABC_recurrence(n, x,
+        2, 0, 1,                            # A_m, B_m, C_m
+        y->1,                               # P_0(y)
```

Comment:
This is not type-stable, same with below functions.

Reply:
see comit [8a24663](https://github.com/JuliaMath/SpecialFunctions.jl/pull/175/commits/8a24663deaf14054facbd3f0158fd83688e60f02). 
all variables are now type stable (always of same type as `x`). the functions as arguments `p0,p1` return either constant values of type `x` or of type as the input argument (which will be `x` itself). 
what do you think?

Reply:
Looks good, just add the `@inferred` in the tests so we know it works correctly

Reply:
done. see commit [cf15a29](https://github.com/JuliaMath/SpecialFunctions.jl/pull/175/commits/cf15a29d31f6bd8d20c32adce56ee60b95cfb7d8)